yarn-project/aztec-rpc/src/aztec_rpc_server/aztec_rpc_server.ts

```diff
@@ -136,14 +114,13 @@ export class AztecRPCServer implements AztecRPC {
   }
 
   /**
-   * Retrieves the list of Aztec addresses associated with the current accounts in the key store.
+   * Retrieves the list of Aztec addresses added to this rpc server
    * The addresses are returned as a promise that resolves to an array of AztecAddress objects.
    *
    * @returns A promise that resolves to an array of AztecAddress instances.
    */
   public async getAccounts(): Promise<AztecAddress[]> {
-    const accounts = this.synchroniser.getAccounts();
-    return await Promise.all(accounts.map(a => a.getAddress()));
+    return await this.db.getAccounts();
```

Comment:
Heads up that this would now return all address/pubkeys registered on the RPC server, which may be more than the ones for which we have a private key. Question is: if I register a recipient's address/partialAddress/pubKey in my RPC server, should I get that address back when I run `getAccounts`?

Reply:
Yeah that's a good point. I feel like we need an API along the lines of `registerPublicKey` which is purely for encrypting notes for others and isn't an 'account' as such.

Reply:
#1067